MainForm.java

```diff
@@ -0,0 +1,545 @@
+		lblResult.setBounds(307, 239, 40, 27);
+		lblResult.setOpaque(true);
+		lblResult.setBackground(new Color(153,153,153));
+		panelInOut.add(lblResult);
+		
+		
+		// Button OK
+		JButton btnOK1 = new JButton("OK");
+		btnOK1.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				
+				// Kiểm tra dữ liệu đầu vào
+				
+				// Biến kiểm tra xem có lỗi hay không
+				boolean check = true;
+				
+				// Nếu dữ liệu đầu vào rỗng thì bắt lỗi
+				if(tfIn.getText().equals("")) {
+					check = false;
+				}
+				
+				// Nếu dữ liệu đầu vào khác 3 kí tự R, G, B thì bắt lỗi
+				char[] input = tfIn.getText().toCharArray();
+				for(int i = 0; i < input.length; i++) {
+					if(input[i] != 'R' && input[i] != 'G' && input[i] != 'B') {
+						check = false;
+						break;
+					}
+				}
+				
+				// Sau khi kiểm tra các trường hợp lỗi nếu có lỗi thì in ra thông báo
+				if(check == false) {
+					try {
+						throw new Exception("Dữ liệu vào không hợp lệ\nDữ liệu vào chỉ chứa các kí tự R, G hoặc B") ;
+					}catch(Exception e1) {
+						JOptionPane.showMessageDialog(rootPane, e1.getMessage());
+					}
+				}else {
+					
+					// Nếu không có lỗi thì thực hiện lấy dữ liệu vào và xử lý 
+					InOut io = new InOut();
+					io.setInput(tfIn.getText());
+//					io.Triangle();
+					io.Triangle2();
+					io.setDTime(Calendar.getInstance().getTime());
+					tfOut.setText(io.getOutput());
+					
+					// Nếu biến kiểm tra Cập nhật = true thì thực hiện việc cập nhật
+					if(checkUpdate == true) {
+						int k = table.getSelectedRow(); // lấy chỉ số dòng được chọn
+						model = (DefaultTableModel) table.getModel();
+						
+						// Lấy thông tin dòng được chọn
+						InOut io1 = new InOut();
+						io1.setInput(model.getValueAt(k, 1).toString());
+						io1.setOutput(model.getValueAt(k, 2).toString());
+						try {
+							io1.setDTime(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse(model.getValueAt(k, 3).toString()));
+						} catch (ParseException e1) {
+							e1.printStackTrace();
+						}
+						
+						// Chạy thực hiện lại với Input vừa được cập nhật
+						InOut io2 = new InOut();
+						io2.setInput(tfIn.getText());
+//						io2.Triangle();
+						io2.Triangle2();
+						io2.setDTime(Calendar.getInstance().getTime());
+						tfOut.setText(io2.getOutput());
+						
+						// Thực hiện việc Cập nhật vào cơ sở dữ liệu
+						if(new ConnectData().updateInOut(io1, io2) == true) {
+							JOptionPane.showMessageDialog(rootPane, "Update Success!"); // In ra "Update Success!" nếu cập nhật thành công
+						}
+						else JOptionPane.showMessageDialog(rootPane, "Not Success!");  // In ra "Not Success!" nếu cập nhật thất bại
+						
+						model.setRowCount(0); // Xóa hết thông tin trong bảng hiện có 
+						
+						ShowData();  // Hiển thị lại dữ liệu vừa cập nhật
+						
+						checkUpdate = false;  // Đặt biến kiểm tra cập nhật về false
+						
+						tabbedPane.setSelectedIndex(2); // Quay trở lại tab History

```

Comment:
Có quá nhiều comment: 
      - Xóa commnet không cần thiết line 284, 286.
      - Nên sửa lại tên biến và tên hàm thay vì comment. 
      - Riêng index = 2 tức là sẽ quay lại tab History, thì chúng ta có thể define 1 biến ở đầu hàm ví dụ: indexTableHistory = 2, và phía dưới gọi nó thi sẽ không cần comment. Và sau này nếu ta có thêm 1 màn hình khác mà index của table history bị đổi thì chỉ cần đổi biến phia trên mà không cần phải đọc lại toàn bộ code để tìm đổi.